get_deps.py

```diff
@@ -30,24 +37,23 @@ def main():
 		if dll_name.startswith('???'):
 			print('Unknown DLL?')
 			continue
-		try:
+		if has_cygpath:
 			# ldd on msys2 gives Unix-style paths, but Python wants them Windows-style
-			# Try cygpath to convert them
+			# If we have cygpath, convert the paths
 			dll_path = subprocess.check_output(['cygpath', '-w', dll_path]).strip()
-		except:
-			# cygpath doesn't exist or failed. Carry on and hope for the best
-			pass
 		if dll_path.lower().startswith('c:\\windows'):
 			print('Skipping system DLL %s' % dll_path)
 			continue
 
-		print('Copying %s...' % dll_path)
-		try:
-			shutil.copyfile(dll_path, os.path.join(args.dest, dll_name))
-		except shutil.Error as err:
-			# This exception only gets raised if both files are exactly the same, but
-			# that doesn't matter to us
-			print('Copying failed: %s' % err)
+		dest_path = os.path.join(args.dest, dll_name)
+		if os.path.normcase(os.path.realpath(dll_path)) == os.path.normcase(os.path.realpath(dest_path)):

```

Comment:
Why not https://docs.python.org/2/library/os.path.html#os.path.samefile ?

Reply:
Unix only, sadly.

Reply:
I'm stupid. yes.